src/pyatsimagebuilder/utils.py

```diff
@@ -70,6 +85,42 @@ def git_clone(url, path, commit_id=None, rm_git=False):
     return hexsha
 
 
+def clone_with_credentials(url, path, credentials):
+    from urllib.parse import urlparse
+    url = urlparse(url)
+    url = url._replace(netloc='{}:{}@{}'.format(
+        credentials['username'], credentials['password'], url.netloc))
+    return git.Repo.clone_from(url.geturl(), path)
+
+
+def clone_with_ssh(url, path, ssh_key):
+
+    # make temp file for ssh_key
+    temp = tempfile.NamedTemporaryFile(mode="w")
+
+    # format the start of the id_rsa file
+    if ssh_key.startswith("-----BEGIN OPENSSH PRIVATE KEY-----"):
+        ssh_key = ssh_key.replace("-----BEGIN OPENSSH PRIVATE KEY-----",
+                                  "-----BEGIN OPENSSH PRIVATE KEY-----\n", 1)
+    else:
+        ssh_key = "-----BEGIN OPENSSH PRIVATE KEY-----\n" + ssh_key
+
+    # format the end of the id_rsa file
+    if ssh_key.endswith("-----END OPENSSH PRIVATE KEY-----"):
+        ssh_key = ssh_key.replace("-----END OPENSSH PRIVATE KEY-----",
+                                  "\n-----END OPENSSH PRIVATE KEY-----\n", 1)
+    else:
+        ssh_key = ssh_key + "\n-----END OPENSSH PRIVATE KEY-----\n"
+
+    temp.write(ssh_key)
+    temp.seek(0)
+
+    os.environ['GIT_SSH_COMMAND'] = 'ssh -i {}'.format(temp.name)
+    repo = git.Repo.clone_from(url, path)
+    del os.environ['GIT_SSH_COMMAND']
```

Comment:
If `GIT_SSH_COMMAND` is already defined, it should be reset to its initial value.

Reply:
it resets to the old one if existed. and deletes otherwise